tools/test/examples/examples_lib.py

```diff
@@ -198,7 +201,7 @@ def print_message(message, name):
                 # list of valid combinations to work through
                 for target, ide in target_cross_ide(ides,
                                                     example['features'],
-                                                    example['targets']):
+                                                    example['targets'] or targets):
```

Comment:
I'm pretty sure this needs to be a set intersection instead of an OR since both are valid filters.